### DISCUSSION TITLE
Saving images corrupts x and y tick labels

### DISCUSSION THREAD

**Discussion by mattharrison** [mwaskom/seaborn#3154] [Q&A]
I'm trying to create some Seaborn plots for an upcoming book and the x/y tick values are getting corrupted when saving it as a PNG.

Using version 0.12.1.

```
# bug in seaborn objects matplotlib saving
import matplotlib.pyplot as plt
import pandas as pd
import seaborn.objects as so
url = 'https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/csv/DAAG/possum.csv'

df = pd.read_csv(url, index_col=[0])
fig, ax = plt.subplots(figsize=(6,4))
(so
 .Plot(df, x='age', y='skullw', color='sex')
 .add(so.Dots())
 .on(fig)
 .plot()
)
fig.savefig('/tmp/so.png', dpi=600, bbox_inches='tight')

```

The result looks like this:
![so](https://user-images.githubusercontent.com/119405/203649304-40184ce4-e661-4e6f-af57-a5a5600053a3.png)

However, if I remove the saving logic....
```
(so
 .Plot(df, x='age', y='skullw', color='sex')
 .add(so.Dots())
)
```

The plot looks like this:

![image](https://user-images.githubusercontent.com/119405/203649447-189cbc15-e57e-4a66-ba0b-6e2480e79f89.png)

**Comment by mwaskom** [comment]
Bear in mind the documentation for [`Plot.on`](https://seaborn.pydata.org/generated/seaborn.objects.Plot.on.html):

> Matplotlib object to use. Passing [matplotlib.axes.Axes](https://matplotlib.org/stable/api/axes_api.html#matplotlib.axes.Axes) will add artists without otherwise modifying the figure. Otherwise, subplots will be created within the space of the given [matplotlib.figure.Figure](https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure) or [matplotlib.figure.SubFigure](https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.SubFigure).

What you're doing is creating a figure anad axes, then passing the _figure_ to `Plot`, so it creates a new set of axes overlapping the one you already created.

The solution is either (a) initialize only the figure with pyplot (with `plt.figure()`) or pass `ax` to `Plot.on`. Note that your choice will also the themetheme, since rc params are cached on the Axes at its time of creation.

PS if your only goal here is to save a png and you aren't doing any other post-processing, why bother with pyplot when you could use [`Plot.save`](https://seaborn.pydata.org/generated/seaborn.objects.Plot.save.html)?

**Comment by mattharrison** [comment]
I didn't know that ``.save`` existed. 

My goal is to export the image and this is the process that I've always used for matplotlib derived tooling (previous Seaborn, Pandas, Yellowbrick, scikit-plot, shap, scikit-learn, etc).

I wouldn't expect saving the figure to mess anything up if there was an existing axes. I will use ``plt.figure`` instead of ``plt.subplots``. Thanks!

**Comment by mwaskom** [comment]
Right, saving the figure isn't messing anything up. Remove the save bit and you will see that it's all due to which object you give `Plot.on`.

> the process that I've always used for matplotlib derived tooling

Why make it more complicated than necessary though?
